### PR TITLE
Use assigned UUID rather than self-generated UUID

### PR DESCRIPTION
**Pull request by shuo-wu**

### CODE REVIEW

File: engine/engine_launcher.go [hunk 1]
```diff
@@ -73,7 +72,7 @@ type Launcher struct {
 
 func NewEngineLauncher(spec *rpc.EngineSpec) (*Launcher, *Engine) {
 	el := &Launcher{
-		UUID:         uuid.NewV4().String(),
+		UUID:         spec.Uuid,
```
Comment (yasker) [inline engine/engine_launcher.go]: In fact it should be spelled as `UUID` since it's short for universally unique identifier

Comment (shuo-wu) [reply engine/engine_launcher.go]: This name is generated by protoc. If we rename `uuid` to `u_u_i_d` in the pb file, we can get `UUID`

**Comment by yasker** [comment]
Should we request UUID if the user wants to operate on the object? To make sure it's operating on the right object.

**Comment by shuo-wu** [comment]
Longhorn engine will call frontend callback and it has no idea about `UUID`.

**Comment by yasker** [comment]
> Longhorn engine will call frontend callback and it has no idea about `UUID`.

It knows. Launcher ID was UUID before.

**Comment by yasker** [comment]
We can do it later though. Not a priority at the moment.

**Comment by shuo-wu** [comment]
> We can do it later though. Not a priority at the moment.

But we need to modify the grpc definition. Hence it cannot be delayed.

**Pull request merged**